galley/pkg/config/analysis/analyzer.go

```diff
@@ -14,41 +14,89 @@
 
 package analysis
 
-import "istio.io/istio/galley/pkg/config/scope"
+import (
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/scope"
+)
 
 // Analyzer is an interface for analyzing configuration.
 type Analyzer interface {
-	Name() string
+	Metadata() Metadata
 	Analyze(c Context)
 }
 
+// Metadata represents metadata for an analyzer
+type Metadata struct {
+	name            string
+	collectionsUsed map[collection.Name]bool
```

Comment:
Can this be simply collection.Names?

If not, and if you need set semantics, then it is more idiomatic (and efficient) to use map[collection.Name]struct{}

Reply:
Yeah it's not really needed here. We're already de-duping in combineCollectionsUsed and again in getUpstreamCollections. I guess we don't really need to de-dupe when combining either. 

I'm surprised that `map[foo]struct{}` is considered more idiomatic than `map[foo]bool`, they're both ugly but the former is even uglier. :)